casper/src/main/scala/coop/rchain/casper/genesis/Genesis.scala

```diff
@@ -139,14 +139,21 @@ object Genesis {
     )
 
   private def fromLine(line: String): Either[String, Vault] = line.split(",") match {
-    case Array(ethAddress, initRevBalanceStr, _) =>
+    case Array(ethAddressString, initRevBalanceStr, _) =>
       Try(initRevBalanceStr.toLong) match {
         case Success(initRevBalance) =>
-          Right(Vault(RevAddress.fromEthAddress(ethAddress), initRevBalance))
+          if (initRevBalance < 0) {
+            RevAddress.fromEthAddress(ethAddressString) match {
+              case Some(ethAddress) => Right(Vault(ethAddress, initRevBalance))
```

Comment:
you can use .asRight here instead of the match:

```
RevAddress.fromEthAddress(ethAddressString)
.map(Vault(_, initiRevBalance))
.asRight(s"Ethereum address $ethAddressString is invalid.")
```